src/Microsoft.AspNetCore.Mvc.Core/Internal/ControllerArgumentBinder.cs

```diff
@@ -163,10 +163,11 @@ public class ControllerArgumentBinder : IControllerArgumentBinder
                 parameter.BindingInfo,
                 parameter.Name);
 
-            if (parameter.BindingInfo?.BinderModelName != null)
+            var parameterModelName = parameter.BindingInfo?.BinderModelName ?? metadata.BinderModelName;
```

Comment:
The `TryUpdateModel` code in `ModelBindingHelper` applies the same logic and will need to be updated